src/lib.rs

```diff
@@ -192,61 +192,113 @@ pub fn webp_load_rgb_from_memory(buf: &[u8]) -> ImageResult<RgbImage> {
     Ok(ImageBuffer::from_raw(width, height, buf.to_vec()).unwrap())
 }
 
-pub fn webp_write<W: Write>(img: &DynamicImage, w: &mut W) -> ImageResult<()> {
-    match img {
-        DynamicImage::ImageRgb8(img) => webp_write_rgb(img, w),
-        DynamicImage::ImageRgba8(img) => webp_write_rgba(img, w),
-        DynamicImage::ImageBgr8(img) => webp_write_bgr(img, w),
-        DynamicImage::ImageBgra8(img) => webp_write_bgra(img, w),
-        DynamicImage::ImageLuma8(_) => webp_write_rgb(&img.to_rgb8(), w),
-        DynamicImage::ImageLumaA8(_) => webp_write_rgba(&img.to_rgba8(), w),
-        DynamicImage::ImageRgb16(_) => webp_write_rgb(&img.to_rgb8(), w),
-        DynamicImage::ImageRgba16(_) => webp_write_rgba(&img.to_rgba8(), w),
-        DynamicImage::ImageLuma16(_) => webp_write_rgb(&img.to_rgb8(), w),
-        DynamicImage::ImageLumaA16(_) => webp_write_rgba(&img.to_rgba8(), w),
-    }
-}
-
-pub fn webp_write_rgba<W: Write, C>(img: &ImageBuffer<Rgba<u8>, C>, w: &mut W) -> ImageResult<()>
-where
-    C: Deref<Target = [u8]>,
-{
-    let buf = libwebp::WebPEncodeRGBA(&img, img.width(), img.height(), img.width() * 4, 75.0)
+pub struct WebpEncoder<W: Write> {
```

Comment:
Could it implement [ImageEncoder](https://docs.rs/image/0.23.14/image/trait.ImageEncoder.html)?